inst/scripts/2.4.0/make-hermann-spermatogenesis-data.Rmd

```diff
@@ -0,0 +1,205 @@
+mv $workdir/AdultMouse_Rep3_possorted_genome_bam.bam.1 $workdir/AdultMouse_Rep3_possorted_genome_bam.bam
+bamtofastq --reads-per-fastq=500000000 $workdir/AdultMouse_Rep3_possorted_genome_bam.bam $workdir/FASTQtmp
+mv $workdir/FASTQtmp/Ad-Ms-Total-Sorted_20k_count_MissingLibrary_1_HK2GNBBXX/bamtofastq_S1_L006_I1_001.fastq.gz $workdir/AdultMouseRep3_S1_L001_I1_001.fastq.gz
+mv $workdir/FASTQtmp/Ad-Ms-Total-Sorted_20k_count_MissingLibrary_1_HK2GNBBXX/bamtofastq_S1_L006_R1_001.fastq.gz $workdir/AdultMouseRep3_S1_L001_R1_001.fastq.gz
+mv $workdir/FASTQtmp/Ad-Ms-Total-Sorted_20k_count_MissingLibrary_1_HK2GNBBXX/bamtofastq_S1_L006_R2_001.fastq.gz $workdir/AdultMouseRep3_S1_L001_R2_001.fastq.gz
+```
+
+# Quantifying spliced and unspliced abundances
+
+Next, we run _alevin_ to obtain spliced and unspliced gene-level counts for each gene and cell.
+
+```{bash}
+salmon alevin -l ISR -i $workdir/gencode.vM24.annotation.expanded.sidx \
+-1 $workdir/AdultMouseRep3_S1_L001_R1_001.fastq.gz \
+-2 $workdir/AdultMouseRep3_S1_L001_R2_001.fastq.gz \
+-o $workdir/alevin_out -p 36 \
+--tgMap $workdir/gencode.vM24.annotation.expanded.tx2gene.tsv \
+--chromium --dumpFeatures --expectCells 1850
+```
+
+# Importing quantifications into R
+
+We use _tximeta_ to import the quantification files into R, after loading the linked transcriptome created previously.
+
+```{r}
+tximeta::loadLinkedTxome(file.path(workdir, "gencode.vM24.annotation.expanded.json"))
+txi <- tximeta::tximeta(coldata = data.frame(
+    names = "AdultMouseRep3",
+    files = file.path(workdir, "alevin_out/alevin/quants_mat.gz"), 
+    stringsAsFactors = FALSE
+), type = "alevin")
+```
+
+The txi object contains a single assay ('counts') containing both spliced and unspliced abundances. 
+We need to split this into two matrices, one with spliced and one with unspliced abundances, with corresponding rows. 
+This can be done using the `splitSE()` function from _tximeta_, providing the data frame linking spliced and unspliced gene identifiers that we created above. 
+
+```{r}
+cg <- read.delim(file.path(workdir, "gencode.vM24.annotation.expanded.features.tsv"),
+                 header = TRUE, as.is = TRUE)
+## Rename the 'intron' column 'unspliced' to make assay names compatible with scVelo
+colnames(cg)[colnames(cg) == "intron"] <- "unspliced"
+txis <- tximeta::splitSE(txi, cg, assayName = "counts")
+```
+
+Next, we download cell type labels provided by the data generators and add to the colData of the created object.
+The cloupe file generated by the authors can be obtained from https://data.mendeley.com/datasets/kxd5f8vpt4/1#file-fe79c10b-c42e-472e-9c7e-9a9873d9b3d8 and loaded into the loupe cell browser, from which cell type labels can be exported into a file, here named "Spermatogenesis_Loupe_CellTypes.csv". 
+
+```{r}
+ctp <- read.csv("Spermatogenesis_Loupe_CellTypes.csv", header = TRUE, as.is = TRUE)
+## Keep only Replicate 3
+ctp <- ctp %>% dplyr::filter(grepl("-3", Barcode)) %>%
```

Comment:
Surely there must be some mistake. This `%>%` doesn't look syntatically valid to me.

Reply:
I suppose you mean that you don't want tidyverse code. I've removed it.